generalizing the solution to the 'extract rows with unequal values'

ibah opened the pull request:
The current solution is correct but can be generalized. It works only for arrays having 3 columns and can't be easily extended. I provide a solution that works for any number of columns and seems to be simpler.

Also a side note, `np.logical_and.reduce()` is equivalent to `np.all()`.

rougier commented:
I would keep both solutions since the current one may also deal with char and/or record array.

ibah commented:
I modified the PR to contain both solutions. I also replaced `np.logical_and.reduce()` with `np.all()`, it works for string arrays and record arrays, as you had said.

rougier commented:
Oh thank, I did not checked myself actually !
